airflow/providers/amazon/aws/hooks/sagemaker_notebook.py

```diff
@@ -0,0 +1,153 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Any
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.providers.amazon.aws.utils import trim_none_values
+from airflow.providers.amazon.aws.utils.tags import format_tags
+
+
+class SageMakerNotebookHook(AwsBaseHook):
+    """Interact with Amazon SageMaker to execute notebooks.
+
+    Provide thick wrapper around
+    :external+boto3:py:class:`boto3.client('sagemaker') <SageMaker.Client>`
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        - :class:`airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        kwargs["client_type"] = "sagemaker"
+        super().__init__(*args, **kwargs)
+
+    def create_instance(
```

Comment:
Do you think it is worth exposing all these parameters? Usually we try to expose all mandatory parameters and parameters which we consider as important (very subjective). But here the list is quite long, I'd try to shorten it since you can still pass the parameters using `kwargs`

Reply:
Refactored the hook to remove calling boto3, implemented the streamlined args in the operator.